widget/buffer.go

```diff
@@ -138,17 +155,17 @@ func (e *editBuffer) String() string {
 	return b.String()
 }
 
-func (e *editBuffer) prepend(s string) {
-	e.moveGap(len(s))
-	copy(e.text[e.caret:], s)
+func (e *editBuffer) prepend(caret int, s string) {
+	e.moveGap(caret, len(s))
+	copy(e.text[caret:], s)
 	e.gapstart += len(s)
 	e.changed = e.changed || len(s) > 0
-	e.dump()
+	e.dump(caret)
```

Comment:
Nit: drop these calls to dump, and perhaps even bufferDebug or dump itself. I've not used it since implementing editBuffer.